quilt/tools/command.py

```diff
@@ -67,8 +68,19 @@
 
 LOG_TIMEOUT = 3  # 3 seconds
 
-VERSION = pkg_resources.require('quilt')[0].version
-
+try:
+    VERSION = pkg_resources.require('quilt')[0].version
+except pkg_resources.DistributionNotFound:
+    # Not installed, so we'll get the version manually.
+    try:
+        mydir = os.path.abspath(stack()[0][1])
+        pkgdir = os.path.join('/', *mydir.split(os.path.sep)[:-3])
+        text = open(os.path.join(pkgdir, 'setup.py')).read()
+        text = text.split('\nQUILT_VERSION = ', 1)[1]
```

Comment:
It is super-hacky, but at the time it was written, the goal was to enable invoking main.py from the tests directory so that pytest runs main.py, without an install, and without creating any new files -- although, some of those restrictions may have been made in part simply as questions on the job description.  Also, this was before I was really familiar with the codebase, and before I had a deeper understanding of what's needed.

Calling from the CLI is technically more complete, but it's significantly slower, and with modifications I've added to main.py, isn't absolutely necessary.  The likelihood that there will be a significant difference between doing a syscall of `['quilt'] + quilt_args` vs calling `main.argument_parser()` and `main.main(args)` is fairly small.

I'd be happy to either store the version in a separate version file and read that, or simply forego actual CLI calls altogether in favor of simulated CLI calls.

Reply:
command.py and setup.py now read from a new VERSION file.  This seemed the least-hacky way of enabling running quilt when not installed.  If this isn't preferred, we can just remove the ability to run quilt when not installed.